lib/api/corbado_service.dart

```diff
@@ -1,118 +1,37 @@
-import 'dart:convert';
+import 'dart:io';
 
+import 'package:corbado_frontend_api_client/frontendapi/lib/api.dart';
 import 'package:flutter/material.dart';
-import 'package:http/http.dart' as http;
+import 'package:passkeys/authenticator/passkey_authenticator.dart';
 
 ///Issues api calls to the corbado backend
 class CorbadoService {
-  final String baseUrlAuth;
-  final String projectID;
+  final String _projectID;
   static String origin = "";
-
-  CorbadoService(this.baseUrlAuth, this.projectID) {
-    debugPrint("Using auth endpoint $baseUrlAuth for projectID $projectID");
-  }
-
-  setOrigin(String origin) {
-    CorbadoService.origin = origin;
-  }
-
-  Future<dynamic> handler(String url, Object body) async {
-    debugPrint("Calling $url...");
-    debugPrint("Requestbody: ${jsonEncode(body)}");
-    var errorStr = " Error calling $url\n\n";
-    var resp = await http.post(Uri.parse("$baseUrlAuth$url"),
-        headers: {
-          'Content-Type': 'application/json; charset=UTF-8',
-          'X-Corbado-ProjectID': projectID,
-          'Origin': origin,
-        },
-        body: jsonEncode(body));
-    debugPrint("Response: ${resp.body}");
-
-    if (resp.statusCode == 200) {
-      return jsonDecode(resp.body);
-    } else if (resp.statusCode == 400) {
-      var body = jsonDecode(resp.body);
-      if (resp.body.contains("validation")) {
-        var error = "Validation error\n";
-        var list = body["error"]["validation"];
-        for (var item in list) {
-          error += "${item["field"]}: ${item["message"]}\n";
-        }
-        throw Exception(errorStr + error);
-      }
+  final ApiClient _client;
+  late final PasskeyAuthenticator _authenticator;
+
+  CorbadoService(this._projectID)
+      : _client =
+            ApiClient(basePath: 'https://$_projectID.frontendapi.corbado.io') {
+    _authenticator = PasskeyAuthenticator();
+    if (Platform.isAndroid) {
```

Comment:
Do we still need this? I think we no longer whitelist origins (maybe you check with vincent).